src/App.tsx

```diff
@@ -2,24 +1,48 @@
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [selectedPerPageSelector, setSelectedPerPageSelector] = useState('5');
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const handleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectedPerPageSelector(event.target.value);
```

Comment:
When updating `selectedPerPageSelector`, convert the value to a number to ensure type consistency. You can use `setSelectedPerPageSelector(Number(event.target.value))`.